scalafmt-tests/src/test/resources/dotty/MatchType.stat

```diff
@@ -0,0 +1,83 @@
+#
+<<< simple match type
+type Elem[X] = X match {
+             case String => Char
+             case Array[t] => t
+}
+>>>
+type Elem[X] = X match {
+  case String   => Char
+  case Array[t] => t
+}
+<<< type bound match type
+type Head[X <: Tuple] = X match {
+           case (x1, ?) => 
+           x1
+         }
+>>>
+type Head[X <: Tuple] = X match {
+  case (x1, ?) =>
+    x1
+}
+<<< recursive match type
+type Len[X] <: Int = X match {
+           case Unit => 0
+           case x *: xs => S[Len[xs]]}
+>>>
+type Len[X] <: Int = X match {
+  case Unit    => 0
+  case x *: xs => S[Len[xs]]
+}
+<<< complex match type
+type Concat[X <: Tuple, Y <: Tuple] <: Tuple = X match {
+           case Unit => 
+          Y
+           case x1 *: xs1 => 
+          x1 *: Concat[xs1, Y]
+         }
+>>>
+type Concat[X <: Tuple, Y <: Tuple] <: Tuple = X match {
+  case Unit =>
+    Y
+  case x1 *: xs1 =>
+    x1 *: Concat[xs1, Y]
+}
+<<< long type bound match type
+maxColumn = 40
+===
+type Concat[X <: Tuple, Y <: Tuple] <: Tuple = X match {
+           case Unit => 
+          Y
+           case x1 *: xs1 => 
+          x1 *: Concat[xs1, Y]
+         }
+>>>
+type Concat[
+    X <: Tuple,
+    Y <: Tuple
+] <: Tuple = X match {
```

Comment:
can we add cases with different values of newlines.source (or beforeMultiline)? would like to see a forced newline for unfold, and single line for next case with fold, for instance.

Reply:
I added some more cases, let me know what you think!